### ISSUE TITLE
Custom activation function not training

### ISSUE THREAD

**Issue by phifre** [keras-team/keras#11057]
I am trying to implement my own custom activation function and tested the approach with the already implemented ELU function.
For some reason the rebuild ELU function doesn't train at all. The loss is at zero after the first epoch and the accuracy doesn't improve.

```python
def custom_activation(x):
    cond = K.greater(x, 0)
    return K.switch(cond, x, K.exp(x) - 1)

get_custom_objects().update({'custom_activation': Activation(custom_activation)})
```

I also tried using the Keras API build into Tensorflow and only Tensorflow functions with the same result.

Full code:
```python
from keras.datasets import mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()

x_train = x_train.reshape(x_train.shape[0], 28*28)
x_test = x_test.reshape(x_test.shape[0], 28*28)

from keras.utils import to_categorical

y_train = to_categorical(y_train)
y_test = to_categorical(y_test)

from keras import Sequential
from keras.layers import Dense, Activation, Dropout
from keras.optimizers import SGD
from keras import backend as K
from keras.utils.generic_utils import get_custom_objects

def custom_activation(x):
    cond = K.greater(x, 0)
    return K.switch(cond, x, K.exp(x) - 1)

get_custom_objects().update({'custom_activation': Activation(custom_activation)})

model = Sequential([
    Dense(128, input_shape=x_train.shape[1:]),
    Activation(custom_activation),
    Dense(128),
    Activation(custom_activation),
    Dense(128),
    Activation(custom_activation),
    Dense(128),
    Activation(custom_activation),
    Dense(128),
    Activation(custom_activation),
    Dense(128),
    Activation(custom_activation),
    Dense(128),
    Activation(custom_activation),
    Dense(128),
    Activation(custom_activation),
    Dense(10),
    Activation('sigmoid')
])

model.compile(SGD(lr=0.01), loss='categorical_crossentropy', metrics=['accuracy'])

model.fit(x=x_train, y=y_train,
          validation_data=[x_test, y_test],
          batch_size=64, epochs=5)
```

Switching the `custom_activation` for `'elu'` works fine.

**Comment by abaxi** [comment]
There may be an overflow error when using K.exp(x) if x is large.

I noticed that K.exp(x) results in loss = nan, but if it is replaced with K.exp(x/100000) it doesn't become nan. I don't know how to get around the issue though.

**Comment by jlherren** [comment]
Also note that CPUs and presumably also GPU often implement `exp(x) - 1` as a single instruction. This is mostly for increased accuracy for when `x` is close to zero.

Here's for example the numpy docs about it: https://docs.scipy.org/doc/numpy/reference/generated/numpy.expm1.html

Perhaps this is related to your problem, since you're not using that.
